core/src/main/java/com/sequenceiq/cloudbreak/core/cluster/ClusterManagerUpscaleService.java

```diff
@@ -86,16 +87,26 @@ public void upscaleClusterManager(Long stackId, Map<String, Integer> hostGroupWi
     }
 
     private void handleWaitForHostsTimeout(boolean repair, boolean primaryGatewayChanged, StackView stack, ExtendedPollingResult result) {
-        String message = String.format("Upscaling cluster manager were not successful, waiting for hosts timed out for nodes: %s",
-                result.getFailedInstanceIds());
-        LOGGER.warn(message);
-        if (!repair && !primaryGatewayChanged && targetedUpscaleSupportService.targetedUpscaleOperationSupported(stack)) {
-            instanceMetaDataService.updateInstanceStatuses(result.getFailedInstanceIds(), InstanceStatus.ZOMBIE,
-                    "Upscaling cluster manager were not successful, waiting for hosts timed out");
+        if (CollectionUtils.isNotEmpty(result.getFailedInstanceIds())) {
```

Comment:
qq: why not result.getFailedInstanceIds().isEmpty?

Reply:
this handled null and empty too, in this case result.getFailedInstanceIds().isEmpty would fail with NPE